core/blockchain_reader.go

```diff
@@ -277,7 +277,7 @@ func (bc *BlockChain) GetTransactionLookup(hash common.Hash) (*rawdb.LegacyTxLoo
 	if tx == nil {
 		progress, err := bc.TxIndexProgress()
 		if err != nil {
-			return nil, nil, nil
+			return nil, nil, err
```

Comment:
```suggestion
			// We return no error here, to signal that we don't expect the transaction to
			// be reachable in the future, since no indexing is happening anymore.
			return nil, nil, nil
```

Reply:
Thanks for your guidance, comment added.